File: backend/app/models/resources.py
from __future__ import annotations
from pydantic import BaseModel


class EventItem(BaseModel):
    uid        : str
    type       : str
    reason     : str
    message    : str
    count      : int = 1
    first_time : str | None = None
    last_time  : str | None = None
    regarding  : str
    regarding_namespace: str | None = None
    source_component: str | None = None
    namespace  : str | None = None


class EventsListResponse(BaseModel):
    events: list[EventItem]
    warning_count: int
    total: int


class PolicyRule(BaseModel):
    verbs        : list[str]
    api_groups   : list[str]
    resources    : list[str]
    resource_names: list[str] = []


class RoleItem(BaseModel):
    name       : str
    namespace  : str | None = None
    rules      : list[PolicyRule]
    created_at : str | None = None
    rule_count : int


class RoleListResponse(BaseModel):
    items: list[RoleItem]


class Subject(BaseModel):
    kind      : str
    name      : str
    namespace : str | None = None


class RoleBindingItem(BaseModel):
    name       : str
    namespace  : str | None = None
    role_ref   : str
    role_kind  : str
    subjects   : list[Subject]
    created_at : str | None = None


class RoleBindingListResponse(BaseModel):
    items: list[RoleBindingItem]


class ServiceAccountItem(BaseModel):
    name       : str
    namespace  : str
    secrets    : list[str]
    created_at : str | None = None
    bound_roles: list[str] = []


class ServiceAccountListResponse(BaseModel):
    items: list[ServiceAccountItem]


class ConfigMapItem(BaseModel):
    name       : str
    namespace  : str
    keys       : list[str]
    data       : dict[str, str] = {}
    created_at : str | None = None


class ConfigMapListResponse(BaseModel):
    items: list[ConfigMapItem]


class SecretItem(BaseModel):
    name        : str
    namespace   : str
    secret_type : str
    keys        : list[str]
    created_at  : str | None = None


class SecretListResponse(BaseModel):
    items: list[SecretItem]

File: frontend/src/types/resources.ts
export interface EventItem {
  uid: string;
  type: string;
  reason: string;
  message: string;
  count: number;
  first_time: string | null;
  last_time: string | null;
  regarding: string;
  regarding_namespace: string | null;
  source_component: string | null;
  namespace: string | null;
}

export interface EventsListResponse {
  events: EventItem[];
  warning_count: number;
  total: number;
}

export interface PolicyRule {
  verbs: string[];
  api_groups: string[];
  resources: string[];
  resource_names: string[];
}

export interface RoleItem {
  name: string;
  namespace: string | null;
  rules: PolicyRule[];
  created_at: string | null;
  rule_count: number;
}

export interface RoleListResponse {
  items: RoleItem[];
}

export interface Subject {
  kind: string;
  name: string;
  namespace: string | null;
}

export interface RoleBindingItem {
  name: string;
  namespace: string | null;
  role_ref: string;
  role_kind: string;
  subjects: Subject[];
  created_at: string | null;
}

export interface RoleBindingListResponse {
  items: RoleBindingItem[];
}

export interface ServiceAccountItem {
  name: string;
  namespace: string;
  secrets: string[];
  created_at: string | null;
  bound_roles: string[];
}

export interface ServiceAccountListResponse {
  items: ServiceAccountItem[];
}

export interface ConfigMapItem {
  name: string;
  namespace: string;
  keys: string[];
  data: Record<string, string>;
  created_at: string | null;
}

export interface ConfigMapListResponse {
  items: ConfigMapItem[];
}

export interface SecretItem {
  name: string;
  namespace: string;
  secret_type: string;
  keys: string[];
  created_at: string | null;
}

export interface SecretListResponse {
  items: SecretItem[];
}
